ui/src/chat/ChatInput/ChatInput.tsx

```diff
@@ -224,6 +226,19 @@ export default function ChatInput({
     }
   }, [messageEditor]);
 
+  useEffect(() => {
+    if (messageEditor && !messageEditor.isDestroyed) {
+      // if text is just a valid URL, make it a link.
+      // this is necessary because the editor doesn't
+      // keep focus on mobile in some cases (chrome/android).
+      if (text && isValidUrl(text)) {
+        messageEditor.commands.selectAll();
+        messageEditor.commands.setLink({ href: text });
+        messageEditor.commands.selectTextblockEnd();
+      }
+    }
+  }, [messageEditor, text]);
+
```

Comment:
should this just be in the paste since this won't work if you paste a link in the middle of some text?

Reply:
If you interact with the input it should just convert it to a clickable link anyway. We don't have access to this editor instance in the onPaste within the MessageEditor component itself afaik.

Reply:
ah ok